pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionActioner.java

```diff
@@ -178,16 +178,31 @@ private void startFunction(FunctionRuntimeInfo functionRuntimeInfo) throws Excep
         runtimeSpawner.start();
     }
 
-    private boolean stopFunction(FunctionRuntimeInfo functionRuntimeInfo) {
+    private void stopFunction(FunctionRuntimeInfo functionRuntimeInfo) {
         Function.Instance instance = functionRuntimeInfo.getFunctionInstance();
         FunctionMetaData functionMetaData = instance.getFunctionMetaData();
         log.info("Stopping function {} - {}...",
                 functionMetaData.getFunctionConfig().getName(), instance.getInstanceId());
         if (functionRuntimeInfo.getRuntimeSpawner() != null) {
             functionRuntimeInfo.getRuntimeSpawner().close();
             functionRuntimeInfo.setRuntimeSpawner(null);
-            return true;
         }
-        return false;
+
+        // clean up function package
+        File pkgDir = new File(
+                workerConfig.getDownloadDirectory(),
+                StringUtils.join(
+                        new String[]{
```

Comment:
Do you want to refactor this + create/update place where you determine the file name?